File: DSA-DAILY/25-07-2025/3487. Maximum_Unique_Subarray_Sum_After_Deletion@leetcode-easy.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <climits>
using namespace std;

class Solution {
public:
    int maxSum(vector<int>& nums) {
        int sum=0;
        unordered_map<int,int> freq;
        for(int i=0;i<nums.size();i++){
            if(nums[i]>0&&freq[nums[i]]<1){
                freq[nums[i]]++;
                sum+=nums[i];
            }
        }
        if(sum==0){
            sum=INT_MIN;
            for(int i=0;i<nums.size();i++){
                sum=max(nums[i],sum);
            }
        }
        return sum;
    }
};

int main() {
    vector<int> nums = {4, 2, 4, 5, 6}; // Example input
    Solution sol;
    int result = sol.maxSum(nums);
    cout << "Maximum Unique Subarray Sum After Deletion: " << result << endl;
}

File: DSA-DAILY/25-07-2025/3487. Maximum_Unique_Subarray_Sum_After_Deletion@leetcode-easy.py
from typing import List

class Solution:
    def maxSum(self, nums: List[int]) -> int:
        sum=0
        freq={}
        for num in nums:
            if num>0 and freq.get(num,0)<1:
                sum+=num
                freq[num]=1
        
        if(sum==0):
            sum=max(nums)
        return sum

if __name__ == "__main__":
    nums = [4, 2, 4, 5, 6]  # Example input
    sol = Solution()
    result = sol.maxSum(nums)
    print("Maximum Unique Subarray Sum:", result)